src/WooCommerce.php

```diff
@@ -186,6 +186,29 @@ public static function woocommerce_get_availability($stock, $product) {
     return $stock;
   }
 
+  /**
+   * Cron event callback to remove outdated back-in-stock product metadata.
+   */
+  public static function cron_ensure_back_in_stock() {
+    global $wpdb;
+    $ids = $wpdb->get_col($wpdb->prepare("SELECT ID FROM {$wpdb->posts}
+      INNER JOIN {$wpdb->postmeta} pm ON {$wpdb->posts}.ID = pm.post_id
+      LEFT JOIN {$wpdb->postmeta} moeve ON {$wpdb->posts}.ID = moeve.post_id AND moeve.meta_key LIKE %s
+      WHERE pm.meta_key = %s
+        AND pm.meta_value <= %s
+        AND moeve.meta_id IS NULL", [
```

Comment:
I'd still recommend to use a table alias for the primary table and the back in stock date meta field, too.  It makes the whole query more readable.
```suggestion
    $ids = $wpdb->get_col($wpdb->prepare("SELECT p.ID FROM {$wpdb->posts} p
      INNER JOIN {$wpdb->postmeta} backinstock ON p.ID = backinstock.post_id
      LEFT JOIN {$wpdb->postmeta} moeve ON p.ID = moeve.post_id AND moeve.meta_key LIKE %s
      WHERE backinstock.meta_key = %s
        AND backinstock.meta_value <= %s
        AND moeve.meta_id IS NULL", [
```